### ISSUE TITLE
running tests?

### ISSUE THREAD

**Issue by dimpase** [rwst/gfan06#2]
How does one run tests in the testsuite?

**Comment by dimpase** [comment]
I see `check:` target in the Makefile, but does it work?

**Comment by kiwifb** [comment]
No idea. I am waiting for you to try it :)

**Comment by dimpase** [comment]
Is it because it's 9 months since it was created? :-)

By the way, does not seem to build with gcc 5.4.

**Comment by kiwifb** [comment]
It is a bit hazy why we have created it in the first place. I think that was c++11 issues but the real upstream moved on since then may be you should have a look at the version that Volker tried in the last 48 hours (0.6.2).

**Comment by dimpase** [comment]
you also have a non-Sage directory structure for cdd headers...

**Comment by dimpase** [comment]
the one Volker tried has no traces on how to run tests either. This is what I am after first of all.

**Comment by kiwifb** [comment]
Ok upstream 0.6.2 tarball seems to have the same `check` target. I'll have to try it.

**Comment by dimpase** [comment]
It would be great to get spkg-check for gfan, you know...

**Comment by kiwifb** [comment]
Would! I am looking into this a bit more with almost pure upstream. `-std=c++0x` burk!

**Comment by kiwifb** [comment]
```
Failed tests:
-------------
testsuite/0000InstallationSection FAILED!
testsuite/0001GroebnerFan FAILED!
testsuite/0003GroebnerFanMod3 FAILED!
testsuite/0004GroebnerFanSymmetry FAILED!
testsuite/0007LeadingTerms FAILED!
testsuite/0008PolynomialSetUnion FAILED!
testsuite/0009RenderStairCase FAILED!
testsuite/0013SecondaryVSGfan FAILED!
testsuite/0100SymmetricGfan FAILED!
testsuite/0101TwoVariables FAILED!
Number of succesful tests 35
Number of failed tests 10
```
The details are confusing. From some of them it looks like the input of the tests hasn't been maintained properly? There is a genuine optimization problem which is solved with a flag for gcc but that flag doesn't exist for clang.

**Comment by dimpase** [comment]
Is this on osx+clang?
How do you run them?

Did you try them on Linux?

I could try them on xeon osx.

**Comment by kiwifb** [comment]
Yes OS X + clang. I just got upstream tarball, started a sage shell (8.2.rc1) edited a few files with `#include cdd/...` to remove the `cdd/` bits since sage puts those directly under `include` then ran make. For clang, you have to remove `-fno-guess-branch-probability` from a compilation line. Then I just ran `make check`.

**Comment by dimpase** [comment]
> pure upstream. -std=c++0x burk!

Shouldn't it be `c++11` ?

**Comment by dimpase** [comment]
On Linux, running tests needed
```
LD_LIBRARY_PATH=$SAGE_LOCAL/lib make check
```

**Comment by dimpase** [comment]
on `gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.9) ` and 
```
Linux localhost 4.4.111-12566-g94bdaa2af495-dirty #1 SMP PREEMPT Tue Apr 3 12:06:49 PDT 2018 x86_64 x86_64 x86_64 GNU/Linux
```
I get
```
Failed tests:
-------------
testsuite/0008PolynomialSetUnion FAILED!
Number of succesful tests 44
Number of failed tests 1
```

**Comment by dimpase** [comment]
The same test results as on Ubuntu on the OSX Xeon box with clang from Xcode 9.3.

**Comment by rwst** [comment]
Just noticed Google Inbox put many github mails into Spam. Sorry.

**Comment by dimpase** [comment]
Anyhow, I've sorted the tests thing on https://trac.sagemath.org/ticket/23353
One minor thing to fix before the latter is ready.
